File: 백준/Silver/2839. 설탕 배달/설탕 배달.c
#include <stdio.h>

int main() {
	int n;
	int count = 0;

	scanf("%d", &n);

	while (1) {
		if (n <= 0) {
			break;
		}

		if (n % 5 == 0) {
			count = count + (n / 5);
			n = 0;
		}
		else {
			n = n - 3;
			count++;
		}
	}

	if (n != 0) {
		printf("-1");
	}
	else {
		printf("%d", count);
	}

	return 0;
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n;
        int count = 0;

        n = sc.nextInt();

        while (true) {
            if (n <= 0) {
                break;
            }

            if (n % 5 == 0) {
                count = count + (n / 5);
                n = 0;
            } else {
                n = n - 3;
                count++;
            }
        }

        if (n != 0) {
            System.out.printf("-1");
        } else {
            System.out.printf("%d", count);
        }

        sc.close();
    }
}
